substrate/client/src/in_mem.rs

```diff
@@ -167,6 +192,39 @@ impl Backend {
 			blockchain: Blockchain::new(),
 		}
 	}
+
+	/// Generate and import a sequence of blocks. A user supplied function is allowed to modify each block header. Useful for testing.
+	pub fn generate_blocks<F>(&self, count: usize, edit_header: F) where F: Fn(&mut block::Header) {
+		use backend::{Backend, BlockImportOperation};
+		let info = blockchain::Backend::info(&self.blockchain).expect("In-memory backend never fails");
+		let mut best_num = info.best_number;
+		let mut best_hash = info.best_hash;
+		let state_root = blockchain::Backend::header(&self.blockchain, BlockId::Hash(best_hash))
+			.expect("In-memory backend never fails")
+			.expect("Best header always exists in the blockchain")
+			.state_root;
+		for _ in 0 .. count {
+			best_num = best_num + 1;
+			let mut header = block::Header {
+				parent_hash: best_hash,
+				number: best_num,
+				state_root: state_root,
+				transaction_root: Default::default(),
+				digest: Default::default(),
+			};
+			edit_header(&mut header);
+
+			let mut tx = self.begin_transaction(BlockId::Hash(best_hash)).expect("In-memory backend does not fail");
```

Comment:
i can't see how this could possibly work in general - the polkadot runtime, at least, alters the state regardless of whether there are any transactions; the `transaction_root` must also be valid for blocks to be executed.

Reply:
This is for tests only. And since this is substrate client the code that uses it should not care. I guess we could have more elaborate testing framework under `/polkadot`